File: python/powercards/player.py
import cardlib
import random

class Player(object):
    def __init__(self, name):
        self.name = name
        deck = [cardlib.Estate() for i in range(3)] + [cardlib.Copper() for i in range(7)]
        random.shuffle(deck)
        self.deck, self.hand = deck[:5], deck[5:]
        self.played, self.discard = [], []
        self.actions, self.buys, self.coins = 0, 0, 0

File: scala/powercards_oo/src/main/scala/powercards/Player.scala
package powercards

import powercards.cards.{Copper, Estate}
import util.Random.shuffle

class Player(val name: String) extends Interactive {
  private val initDeck = shuffle(Seq.fill(3)(new Estate) ++ Seq.fill(7)(new Copper))
  var deck: Vector[Card] = initDeck.drop(5).toVector
  var hand: Vector[Card] = initDeck.take(5).toVector
  var played: Vector[Card] = Vector.empty
  var discard: Vector[Card] = Vector.empty

  var actions: Int = 0
  var buys: Int = 0
  var coins: Int = 0
}
